src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -1743,7 +1759,7 @@ internal static bool IsAttributeDefined(RuntimeModule decoratedModule, int decor
 
         #region Private Static Methods
         private static bool AttributeUsageCheck(
-            RuntimeType attributeType, bool mustBeInheritable, object[] attributes, IList derivedAttributes)
+            RuntimeType attributeType, bool mustBeInheritable, RuntimeType.ListBuilder<object> derivedAttributes)
```

Comment:
`ListBuilder` is `struct`. It should be passed around via `ref`.

Reply:
Agreed - looking at this locally. `private static unsafe object[] GetCustomAttributes()` on 1496 needs refactor love though, as the old method is a bit brittle there - seeing about a refactor to make this cleaner and efficient.